spec/models/manageiq/providers/vmware/network_manager/refresher_spec.rb

```diff
@@ -30,205 +63,352 @@
-      )
-      expect(vapp_subnet.network_ports.count).to eq(1)
-      expect(vapp_subnet.vms.count).to eq(1)
-    end
+          def assert_specific_subnet
+            expect(vdc_subnet).to have_attributes(
+              :name                           => 'subnet-ManageIQ Dev external network',
+              :cidr                           => '10.12.0.24/16',
+              :status                         => nil,
+              :dhcp_enabled                   => nil,
+              :gateway                        => "10.12.0.24",
+              :network_protocol               => nil,
+              :dns_nameservers                => ["10.12.0.13"],
+              :ipv6_router_advertisement_mode => nil,
+              :ipv6_address_mode              => nil,
+              :extra_attributes               => nil,
+              :type                           => subnet_type,
+              :ext_management_system          => @ems_network,
+              :availability_zone              => nil,
+              :cloud_network                  => vdc_net,
+              :cloud_tenant                   => nil,
+              :network_router                 => nil,
+              :network_group                  => nil,
+              :parent_cloud_subnet            => nil
+            )
+            expect(vdc_subnet.network_ports.count).to eq(2)
+            expect(vdc_subnet.vms.count).to eq(2)
+          end
 
-    def assert_specific_network_port
-      expect(net_port).to have_attributes(
-        :name                  => 'vAppRHEL7-w-002#NIC#0',
-        :mac_address           => '00:50:56:01:00:0c',
-        :device_type           => 'VmOrTemplate',
-        :source                => 'refresh',
-        :ext_management_system => @ems_network,
-        :device                => vm
-      )
-    end
+          def assert_specific_network_port
+            expect(net_port).to have_attributes(
+              :name                  => 'vm5#NIC#0',
+              :mac_address           => '00:50:56:01:01:28',
+              :device_type           => 'VmOrTemplate',
+              :ext_management_system => @ems_network,
+              :device                => vm,
+            )
+          end
 
-    def assert_specific_network_router
-      expect(router).to have_attributes(
-        :name                  => 'Router RedHat external network -> vApp network test',
-        :type                  => router_type,
-        :cloud_network         => vdc_net,
-        :ext_management_system => @ems_network,
-        :cloud_subnets         => [vapp_subnet]
-      )
-    end
+          def assert_specific_vm_networking
+            expect(vm).to have_attributes(
+              :name           => 'vm5',
+              :mac_addresses  => [net_port.mac_address],
+              :cloud_networks => [vdc_net],
+              :cloud_subnets  => [vdc_subnet],
+              :network_ports  => [net_port],
+              :floating_ips   => []
```

Comment:
Please also check field `:ipaddresses` like we did in the old [refresher spec](https://github.com/ManageIQ/manageiq-providers-vmware/blob/master/spec/models/manageiq/providers/vmware/network_manager/refresher_spec.rb#L204). It should contain a list of  VM's IP addresses - first come local addresses and in the end of list the floating ip address. This list should match what you see on the vCD's Dashboard.

This field is the most important one because it's displayed on GUI. As a matter of fact, field `:ipaddresses` is why we're implementing NetworkManager at all :) Make sure you check it for every VM in every describe.